jadx-core/src/main/java/jadx/core/dex/nodes/utils/CodeNodeUtils.java

```diff
@@ -0,0 +1,85 @@
+package jadx.core.dex.nodes.utils;
+
+import jadx.core.dex.info.AccessInfo;
+import jadx.core.dex.nodes.ClassNode;
+import jadx.core.dex.nodes.FieldNode;
+import jadx.core.dex.nodes.ICodeNode;
+import jadx.core.dex.nodes.MethodNode;
+import jadx.core.dex.nodes.RootNode;
+import jadx.core.utils.exceptions.JadxRuntimeException;
+
+class CodeNodeUtils {
+
+	private final RootNode root;
+
+	CodeNodeUtils(RootNode rootNode) {
+		this.root = rootNode;
+	}
+
+	boolean isAccessible(ICodeNode targetNode, ICodeNode callerNode) {
+		ClassNode targetCls = getDeclaringClass(targetNode);
+		ClassNode callerCls = getDeclaringClass(callerNode);
```

Comment:
I notice that we can have inner class in another maybe private inner class, so it requires to check all upper level visibility and use more strict one as an result/effective one. This will make code more complex, but it will be nice to check.
Anyway, at this point we might need a test case for this method (just a unit test) :slightly_smiling_face:

Reply:
I'm not quite sure what you mean. Members of inner classes are accessible from everywhere inside the top class.

<details><summary>Sample</summary>
<p>

```java
public class TestCls {

  private class A {

    private class AA {

      private void testFromAA() {
        testFromA();

        B b = new B();
        b.testFromB();

        B.BB bb = (new B()).new BB();
        bb.testFromBB();
      }
    }

    private void testFromA() {
      AA aa = new AA();
      aa.testFromAA();

      B b = new B();
      b.testFromB();

      B.BB bb = (new B()).new BB();
      bb.testFromBB();
    }
  }

  private class B {

    private class BB {

      private void testFromBB() {
        testFromB();

        A a = new A();
        a.testFromA();

        A.AA aa = (new A()).new AA();
        aa.testFromAA();
      }
    }

    private void testFromB() {
      BB bb = new BB();
      bb.testFromBB();

      A a = new A();
      a.testFromA();

      A.AA aa = (new A()).new AA();
      aa.testFromAA();
    }
  }

  private void test() {
    A a = new A();
    a.testFromA();

    B b = new B();
    b.testFromB();

    A.AA aa = (new A()).new AA();
    aa.testFromAA();

    B.BB bb = (new B()).new BB();
    bb.testFromBB();
  }
}
```

</p>
</details>

Reply:
```java

public class A {
	public void call() {
		B.BB.BBB.test();
	}
}

public class B {
	private static class BB {
		public static class BBB {
			public static void test() {
			}
		}
	}
}
```
Here A and B different top classes and while BBB is public you can't access it with `B.BB.BBB.test();` because `B.BB has private access in B` (java compiler said this).

But sure, such cases looks exotic, so we might be better to fix these in separate PR.

Reply:
Ok, I'll think about that.